src/deploy/functions/validate.ts

```diff
@@ -8,14 +8,16 @@ import * as fsutils from "../../fsutils";
 import * as backend from "./backend";
 import * as utils from "../../utils";
 import * as secrets from "../../functions/secrets";
+import { serviceForEndpoint } from "./services";
 
 /** Validate that the configuration for endpoints are valid. */
 export function endpointsAreValid(wantBackend: backend.Backend): void {
-  functionIdsAreValid(backend.allEndpoints(wantBackend));
+  const endpoints = backend.allEndpoints(wantBackend);
+  functionIdsAreValid(endpoints);
+  endpoints.forEach((ep) => serviceForEndpoint(ep).validateTrigger(ep as any, wantBackend));
```

Comment:
nit: google style guide forbids `forEach`. Just use `for( of )` instead.

Reply:
👍